### PR TITLE
refactor: products page

### PR DESCRIPTION
**Pull request by bharati-21**
- Added single product page
- Move from cart to wishlist
- Add out of stock overall on products on the product page
- Disable add to cart, add to the wishlist on the out of stock single product page

Deploy Preview - https://deploy-preview-16--bookery.netlify.app/

### CODE REVIEW

File: src/components/Cart/CartListItem.jsx [hunk 1]
```diff
@@ -95,16 +104,58 @@ const CartListItem = ({ cartItem }) => {
...
+	const isItemInWishList = wishListItems.find(
+		(wishListItem) => wishListItem._id === _id
+	);
+
+	const handleMoveItemToWishList = async () => {
+		if (isItemInWishList) {
+			showToast("Item added to wishlist!", "success");
+			handleRemoveFromCart(false);
+		} else {
+			setIsOngoingNetworkCall(true);
```
Comment (rahulyadav139) [inline src/components/Cart/CartListItem.jsx:127]: Appreciation: Great naming convention. Codes are readable and clean. Great work.

File: src/pages/ProductPage/ProductPageItem.jsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        }
+        else {
+            setBookInCart(false);
+        }
+
+        if(isBookInWishList) {
+            setBookInWishList(true);
+        }
+        else {
+            setBookInWishList(false);
```
Comment (rahulyadav139) [inline src/pages/ProductPage/ProductPageItem.jsx:56]: Everything is great but template literal would be great here.

File: src/pages/ProductPage/ProductPageItem.jsx [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        }
+    }, []);
+
+    const outOfStock = !offers.inStock;
+
+
+    const localeOriginalPrice = originalPrice.toLocaleString("en-IN", {minimumFractionDigits: 2, maximumFractionDigits: 2});
+    const localeSellingPrice = sellingPrice.toLocaleString("en-IN", {minimumFractionDigits: 2, maximumFractionDigits: 2});
```
Comment (rahulyadav139) [inline src/pages/ProductPage/ProductPageItem.jsx:64]: I don't think you need to specify minimumFractionDigits: 2, I think it is by default set to 2.

**Review by rahulyadav139 — APPROVED** [review]

File: src/components/Cart/CartListItem.jsx [hunk 4]
```diff
@@ -95,16 +104,58 @@ const CartListItem = ({ cartItem }) => {
...
 		} catch (error) {
 			console.log("Something went really wrong!");
 		}
 	};
 
+	const isItemInWishList = wishListItems.find(
+		(wishListItem) => wishListItem._id === _id
+	);
+
+	const handleMoveItemToWishList = async () => {
```
Comment (ivinayakg) [inline src/components/Cart/CartListItem.jsx:122]: again delegate to another file

**Review by ivinayakg — APPROVED** [review]
The code is good,Good job

File: src/components/Cart/CartListItem.jsx (hunk 1 above)
Comment (bharati-21) [reply src/components/Cart/CartListItem.jsx:127]: Thank you!

File: src/pages/ProductPage/ProductPageItem.jsx (hunk 2 above)
Comment (bharati-21) [reply src/pages/ProductPage/ProductPageItem.jsx:56]: I shall make this change.

File: src/pages/ProductPage/ProductPageItem.jsx (hunk 3 above)
Comment (bharati-21) [reply src/pages/ProductPage/ProductPageItem.jsx:64]: Actually, by default on doing .toLocaleString(), there are no decimals. Hence the use for minimumFractionDigits.

File: src/components/Cart/CartListItem.jsx (hunk 4 above)
Comment (bharati-21) [reply src/components/Cart/CartListItem.jsx:122]: Sure. Thank you

**Comment by bharati-21** [comment]
> The code is good,Good job

Thank you for your review!

**Pull request merged**